File: sdks/dotnet/src/Croniq.Runner.Sdk/Protocol/PollRequest.cs
using System.Text.Json.Serialization;

namespace Croniq.Runner.Sdk.Protocol;

internal sealed record PollRequest(
    [property: JsonPropertyName("runner_id")] string RunnerId,
    [property: JsonPropertyName("capabilities")] IReadOnlyList<string> Capabilities,
    [property: JsonPropertyName("max_inflight")] int MaxInflight,
    [property: JsonPropertyName("inflight")] IReadOnlyList<string> Inflight,
    [property: JsonPropertyName("instance_id"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] string? InstanceId,
    [property: JsonPropertyName("tags")] IReadOnlyList<string> Tags);

File: sdks/java/core/src/main/java/io/croniq/runner/protocol/PollRequest.java
package io.croniq.runner.protocol;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/** POST /v1/work/poll request body. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record PollRequest(
        @JsonProperty("runner_id") String runnerId,
        @JsonProperty("capabilities") List<String> capabilities,
        @JsonProperty("max_inflight") int maxInflight,
        @JsonProperty("inflight") List<String> inflight,
        @JsonProperty("instance_id") String instanceId,
        @JsonProperty("tags") List<String> tags) {}
